src/main_gui.cpp

```diff
@@ -247,18 +247,19 @@ struct MainWindow : Window
 	{
 		this->DrawWidgets();
 		if (_game_mode == GM_MENU) {
+			int scaling_factor = ZOOM_LVL_OUT_4X / _gui_zoom;
```

Comment:
This is wrong, the zoom level is a logarithmic scale not linear, and the most zoomed in (largest scaling factor) has a value of 0 so this would cause division by zero at 4x GUI scale. Use the `UnScaleByZoom` function to scale GUI elements.

Reply:
Thanks for the pointer. I've updated the PR accordingly.